### DISCUSSION TITLE
Write List of all Streampipes Variables in the Wiki and harmozise InteliJ runConfigurations?

### DISCUSSION THREAD

**Discussion by flomickl** [apache/streampipes#1325] [Dev] [answered]
I think we should try to complete the list of all StreamPipes Variables that exists
plus additional Java Variables that can be used in intellij.
Before creating an Issue, we should discuss it here.

The outcome should be documented here: https://cwiki.apache.org/confluence/x/9YoODg

As a result, we also should harmonize the default variables if possible in the runConfigurations files:
https://github.com/apache/streampipes/tree/dev/.idea/runConfigurations

For me, most of these configurations are not running with the CLI by default.
Oder habe nur ich dieses Problem?

**Comment by tenthe** [comment, marked as answer]
Hello @flomickl,
I think it's great that you started to collect all current variables in the wiki.
This is a good step to get an overview.

Once we have collected all the variables, I suggest that we place them in one (or maybe two) places in the code and add the description to the code. We can then reference this class in the wiki.
That way we don't have to update two places when something changes.

I think @dominikriemer also was working on the variables recently, see [1].

[1] https://github.com/apache/streampipes/blob/dev/streampipes-commons/src/main/java/org/apache/streampipes/commons/constants/Envs.java
